webserver/views/api/v1/datasets.py

```diff
@@ -217,7 +262,38 @@ def add_recordings(dataset_id):
 
     :resheader Content-Type: *application/json*
     """
-    raise NotImplementedError
+    ds = get_check_dataset(dataset_id)
+    if ds["author"] != current_user.id:
+        raise api_exceptions.APIUnauthorized("You can't add the recording(s).")
+    class_dict = request.get_json()
+    if not class_dict:
+        raise api_exceptions.APIBadRequest("Data must be submitted in JSON format.")
+    if "class_name" not in class_dict:
+        raise api_exceptions.APIBadRequest("class_name key missing in JSON request.")
+    if class_dict["class_name"] not in (classes["name"] for classes in ds["classes"]):
+        raise api_exceptions.APIBadRequest("Class not present in the dataset.")
+    if "recordings" not in class_dict:
+        raise api_exceptions.APIBadRequest("recordings key missing in JSON request.")
+    for mbid in class_dict["recordings"]:
+        try:
+            UUID(mbid, version=4)
+        except ValueError:
+            raise api_exceptions.APIBadRequest("MBID %s not a valid UUID" % (mbid, ))
+    for classes in ds["classes"]:
+        if classes["name"] == class_dict["class_name"]:
+            for mbid in class_dict["recordings"]:
+                if mbid in (classes["recordings"]):
```

Comment:
You could do this check by turning the two lists of recordings into sets and taking the difference.